### DISCUSSION TITLE
Can't reverse proxy HyperHDR behind Caddy

### DISCUSSION THREAD

**Discussion by LeVraiRoiDHyrule** [awawa-dev/HyperHDR#558] [Q&A]
Hi,

I would like to reverse proxy HyperHDR behind my Caddy server. It doesn't work. HyperHDR loads perfectly when accessed directly on the local network.
When accessed from the outside, HyperHDR display these messages forever:

```
2023-05-04T00:27:21.694 WEBSOCKET              : <DEBUG> WebSocketClient.cpp:30:WebSocketClient() | New connection from ::ffff:172.18.0.7
2023-05-04T00:27:24.508 WEBSOCKET              : <DEBUG> JsonAPI.cpp:97:handleInstanceSwitch() | Client '::ffff:172.18.0.7' switch to HyperHDR instance 0
2023-05-04T00:27:24.507 WEBSOCKET              : <DEBUG> WebSocketClient.cpp:30:WebSocketClient() | New connection from ::ffff:172.18.0.7
2023-05-04T00:27:21.694 WEBSOCKET              : <DEBUG> JsonAPI.cpp:97:handleInstanceSwitch() | Client '::ffff:172.18.0.7' switch to HyperHDR instance 0
```
And the UI doesn't load.

My Caddy server works for every other app except HyperHDR
Could someone help me to find what's wrong ?
Thanks in advance for any answer,
have a great day

**Comment by awawa-dev** [comment]
Hi
You need also setup WebSocket over proxy for HyperHDR ...rather advanced stuff. Never tried it.

**Reply by LeVraiRoiDHyrule** [reply]
Hi, sorry for the late reply.
Caddy websockets are enabled by default, there is no need to set them up. I have a lot of other apps that use websockets and there is nothing special to do with Caddy.
I'm not sure what the problem is but it seems to come more from HyperHDR.

**Reply by LeVraiRoiDHyrule** [reply]
Hi,
Did you try reverse proxying Hyperhdr? Maybe on a different proxy like Nginx? If yes, did it work and with what configuration?
Thanks in advance and have a great day

**Reply by awawa-dev** [reply]
Hi
No, I haven't tried HyperHDR with a proxy. You probably need to debug the HyperHDR javascript to find out what is causing it and if possible, what needs to be configured differently in the proxy setup. I think the main suspect is host address detection or local socket handling by proxy when the site is accessed through a reverse proxy. Don't use or support it, so unfortunately I can't help with that.

**Comment by skbtwiz** [comment]
Hello,

I have a similar set up as the original post. I have a Pi zero running Hyperhdr and a Pi4b running an nginx proxy manager docker container managing the reverse proxy of many other containers on the pi4b as well as the hyperhdr instance on the Pi zero. I came across this thread because the websockets suddenly stopped working today after a recent upgrade to the new raspbian OS. They have worked flawlessly so far and I commend the developer on building a robust app. All I had to do to fix it was reboot both the Pis a bunch of times until the problem just somehow went away. This makes me think that it was more an issue on the reverse proxy side of things, as ssh via reverse proxy was not working either but the hyperhdr instance worked when accessed through the IP address. It might be an issue with docker network bridges getting backlogged with junk data, that's my preliminary diagnosis. If OP is using docker they should look into resetting their environment.

Thank you, hope this helps. By my extensive testing I can confirm Hyperhdr works out of the box with reverse proxies as long as websockets are enabled. I thank the developer for their time and effort into building this app.

**Comment by I-Would-Like-To-Report-A-Bug-Please** [comment]
Huh, got the same thing. Everything works except for HyperHDR.
